components/dashboard/src/projects/ProjectSettings.tsx

```diff
@@ -56,19 +55,14 @@ export default function () {
     const { teams } = useContext(TeamsContext);
     const team = getCurrentTeam(location, teams);
 
-    const [isLoading, setIsLoading] = useState<boolean>(true);
-    const [isIncrementalPrebuildsEnabled, setIsIncrementalPrebuildsEnabled] = useState<boolean>(false);
-    const [isPersistentVolumeClaimEnabled, setIsPersistentVolumeClaimEnabled] = useState<boolean>(false);
     const [isShowPersistentVolumeClaim, setIsShowPersistentVolumeClaim] = useState<boolean>(false);
+    const [projectSettings, setProjectSettings] = useState<ProjectSettings>({});
 
     useEffect(() => {
         if (!project) {
             return;
         }
-        setIsLoading(false);
-        setIsIncrementalPrebuildsEnabled(!!project.settings?.useIncrementalPrebuilds);
-        setIsPersistentVolumeClaimEnabled(!!project.settings?.usePersistentVolumeClaim);
-
+        setProjectSettings({ ...project.settings });
```

Comment:
What is the benefit of having an extra `projectSettings` state here, given that we already have a `project`? 🤔

Couldn't we use `project.settings` as the source of truth? (And modify that + call `updateProjectSettings` when things are toggled?)

Reply:
Because we are using react state and need to update the object for react to rerender. react doesn't check deep references

Reply:
Ah, that's a good point, thanks.

So I guess getting `setProject` from the ProjectContext, and calling `setProject(project)` after modifying the `project.settings` wouldn't actually update the state, right? 🤔

In any case, the current implementation is fine by me assuming it works 😊 (will test it soon) so this is not a blocker.

Reply:
Yes, it does not update the state.